backend/routes/api/spots.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const { Spot } = require('../../db/models');
+const { requireAuth } = require('../../utils/auth');
+const { check, validationResult } = require('express-validator');
+
+const router = express.Router();
+
+const validateSpot = [
+  check('address').exists({ checkFalsy: true }).withMessage('Address is required.'),
+  check('city').exists({ checkFalsy: true }).withMessage('City is required.'),
+  check('state').exists({ checkFalsy: true }).withMessage('State is required.'),
+  check('country').exists({ checkFalsy: true }).withMessage('Country is required.'),
+  check('lat').isFloat({ min: -90, max: 90 }).withMessage('Latitude must be between -90 and 90.'),
+  check('lng').isFloat({ min: -180, max: 180 }).withMessage('Longitude must be between -180 and 180.'),
+  check('name').exists({ checkFalsy: true }).withMessage('Name is required.'),
+  check('description').exists({ checkFalsy: true }).withMessage('Description is required.'),
+  check('price').isFloat({ gt: 0 }).withMessage('Price must be greater than 0.'),
+];
+
+router.post('/', requireAuth, validateSpot, async (req,res) => {
+    const { user } = req;
+    const { address , city , state, country , lat, lng, name, description, price } = req.body;
+    const validationErrors = validationResult(req);
```

Comment:
Why are you using validationResult?

Reply:
I was using ValidationResult from the express - validator to be able to handle and check the results of those validations checks that I defined in my validateSpot middleware

Reply:
Can you provide the docs you referenced for this, and explain why we may not want to use this function?

Reply:
https://docs.greatexpectations.io/docs/reference/learn/terms/validation_result/#:~:text=A%20Validation%20Result%20is%20an,Validates